app/src/main/java/com/github/sdp/ratemyepfl/fragment/review/RoomReviewInfoFragment.kt

```diff
@@ -20,20 +29,82 @@ class RoomReviewInfoFragment : Fragment(R.layout.fragment_room_review_info) {
     private val viewModel by activityViewModels<ClassroomInfoViewModel>()
 
+    private lateinit var roomNoiseInfoTextView: TextView
+    private lateinit var noiseMeasureButton: ImageButton
+
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
         viewModel.room.observe(viewLifecycleOwner) {
             view.findViewById<TextView>(R.id.roomIdInfo).text = it?.toString()
         }
-        viewModel.numReviews.observe(viewLifecycleOwner){
+        viewModel.numReviews.observe(viewLifecycleOwner) {
             view.findViewById<TextView>(R.id.roomNumReview).text =
                 getNumReviewString(requireContext(), it)
         }
-        viewModel.averageGrade.observe(viewLifecycleOwner){
+        viewModel.averageGrade.observe(viewLifecycleOwner) {
             view.findViewById<RatingBar>(R.id.roomRatingBar).rating = it.toFloat()
         }
+
+        roomNoiseInfoTextView = view.findViewById(R.id.roomNoiseInfoTextView)
+        viewModel.noiseData.observe(viewLifecycleOwner) {
+            displayRoomNoise(it)
+        }
+
+        // Record audio
+        val audioPermissionLauncher =
+            PermissionUtils.requestPermissionLauncher({ startAudio() }, this, requireContext())
+
+        noiseMeasureButton = view.findViewById(R.id.noiseMeasureButton)
+        noiseMeasureButton.setOnClickListener {
+            PermissionUtils.startPhoneFeature(
+                { startAudio() },
+                audioPermissionLauncher,
+                requireContext(),
+                Manifest.permission.RECORD_AUDIO
+            )
+        }
+
+    }
+
+    private fun displayRoomNoise(noiseData: Map<String, Int>) {
+        if (noiseData.isEmpty()) return
+
+        val sortedMap = noiseData.toSortedMap()
+        val mostRecentDate = sortedMap.lastKey()
+        val mostRecentMeasure = sortedMap[mostRecentDate]
+
+        val (text, color) = SoundDisplayUtils.decibelMap(mostRecentMeasure!!)
+        roomNoiseInfoTextView.text =
+            getString(
+                R.string.room_noise_info,
+                text,
+                mostRecentMeasure.toString(),
+                TimeUtils.prettyTime(mostRecentDate)
+            )
+        roomNoiseInfoTextView.setTextColor(color)
+    }
+
+
+    /**
+     * Gets the noise intensity that was measured by the user in [AudioRecordActivity]
+     */
+    private fun startAudio() {
+        val intent = Intent(requireContext(), AudioRecordActivity::class.java)
+        getAudioResults.launch(intent)
     }
 
+    private val getAudioResults =
+        registerForActivityResult(ActivityResultContracts.StartActivityForResult()) { result ->
+            if (result.resultCode == Activity.RESULT_OK) {
+                val data: Intent? = result.data
+                val measure = data?.getDoubleExtra(AudioRecordActivity.EXTRA_MEASUREMENT_VALUE, 0.0)
+                if (measure != null && measure > 0.0) {
```

Comment:
You check here that the measure is not 0 before uploading, but in the AudioRecordActivity you make a toast saying that the measure was successfully, even if the measure is 0.

Reply:
My bad, it should be possible to upload zero measures well. I will remove this strict condition.